src/components/views/OfflineSettings.vue

```diff
@@ -0,0 +1,376 @@
+const isSubmitting = ref(false);
+
+const updateRefsFromUserData = (newUserData) => {
+  // Ensure newUserData and its properties are defined
+  const offlineAdministrations = newUserData?.offlineAdministrations ?? [];
+  const offlineTasks = newUserData?.offlineTasks ?? [];
+
+  // Assign values to selectedOfflineAdministrations and selectedOfflineTasks
+  selectedOfflineAdministrations.value = offlineAdministrations.length > 0 ? [...offlineAdministrations] : [];
+  selectedOfflineTasks.value = offlineTasks.length > 0 ? [...offlineTasks] : [];
+  offlineEnabled.value = newUserData?.offlineEnabled ?? false;
+};
+
+watch(userData, (newUserData) => {
+  updateRefsFromUserData(newUserData);
+});
+
+const addOfflineAdministration = () => {
+  if (selectedOfflineAdministrations.value.includes(selectedOfflineAdministration.value)) {
+    toast.add({
+      severity: 'info',
+      summary: 'Task already added',
+      detail: 'This task has already been added',
+      life: 3000,
+    });
+    return;
+  } else {
+    selectedOfflineAdministrations.value.push(selectedOfflineAdministration.value);
+    return;
+  }
+};
+
+const addOfflineTask = () => {
+  if (selectedOfflineTasks.value.includes(selectedOfflineTask.value)) {
+    toast.add({
+      severity: 'info',
+      summary: 'Task already added',
+      detail: 'This task has already been added',
+      life: 3000,
+    });
+    return;
+  } else {
+    selectedOfflineTasks.value.push(selectedOfflineTask.value);
+    return;
+  }
+};
+
+const removeOfflineAdministration = (name) => {
+  console.log('called offline administrations');
+  selectedOfflineAdministrations.value = selectedOfflineAdministrations.value.filter((task) => task !== name);
+};
+
+const removeOfflineTask = (name) => {
+  console.log('called delete offline on:', name);
+  selectedOfflineTasks.value = selectedOfflineTasks.value.filter((task) => task !== name);
+};
+
+unsubscribe = authStore.$subscribe(async (mutation, state) => {
+  if (state.roarfirekit.restConfig) init();
+});
+
+onMounted(() => {
+  if (roarfirekit.value.restConfig) init();
+  updateRefsFromUserData(userData.value);
+});
+
+const saveOfflineSettings = async () => {
+  isSubmitting.value = true;
+  await roarfirekit.value
+    .updateUserData(uid.value, {
+      offlineEnabled: offlineEnabled.value,
+      offlineTasks: selectedOfflineTasks.value,
+      offlineAdministrations: selectedOfflineAdministrations.value,
+    })
+    .then(() => {
+      isSubmitting.value = false;
+      toast.add({ severity: 'success', summary: 'Updated', detail: 'Your Info has been updated', life: 3000 });
+    })
+    .catch((error) => {
+      console.error('Error updating user data', error);
+      toast.add({
+        severity: 'error',
+        summary: 'Unexpected Error',
+        detail: 'An unexpected error has occurred.',
+        life: 3000,
+      });
+    });
+  // invalidate tanstack queries
+
+  queryClient.invalidateQueries(['userData', uid]);
```

Comment:
```suggestion
await queryClient.invalidateQueries(['userData', uid]);
```
Either `await` this or move it into the .then() block above (don't think it matters which one).